suite-common/wallet-core/src/send/sendFormCardanoThunks.ts

```diff
@@ -114,44 +126,34 @@ export const composeCardanoSendFormTransactionThunk = createThunk(
             }
 
             const feeLabel = predefinedLevels[index].label;
-            wrappedResponse[feeLabel] = tx;
+            resultLevels[feeLabel] = tx;
         });
 
-        return wrappedResponse;
+        return resultLevels;
     },
 );
 
-export const signCardanoSendFormTransactionThunk = createThunk(
+type SignCardanoTransactionThunkArguments = Omit<
+    SignTransactionThunkArguments,
+    'formState' | 'precomposedTransaction' | 'accountStatus'
+> & {
+    precomposedTransaction: PrecomposedTransactionFinalCardano;
+};
+
+export const signCardanoSendFormTransactionThunk = createThunk<
+    { serializedTx: string },
+    SignCardanoTransactionThunkArguments,
+    { rejectValue: SignTransactionError }
+>(
     `${SEND_MODULE_PREFIX}/signCardanoSendFormTransactionThunk`,
-    async (
-        {
-            precomposedTransaction,
-            selectedAccount,
-        }: {
-            precomposedTransaction: PrecomposedTransactionFinalCardano;
-            selectedAccount?: Account;
-        },
-        { dispatch, getState, extra },
-    ) => {
-        const {
-            selectors: { selectSelectedAccountStatus },
-        } = extra;
-
-        const selectedAccountStatus = selectSelectedAccountStatus(getState());
-        const device = selectDevice(getState());
-
-        if (
-            G.isNullable(selectedAccount) ||
```

Comment:
this check is no more needed?

Reply:
this check was moved to the `signTransactionThunk` before passing to this thunk. I did it again to remove the code duplication.